iib/workers/tasks/build.py

```diff
@@ -0,0 +1,171 @@
+    released.
+
+    :param str dockerfile_dir: the path to the directory containing the data generated by the
+        opm command.
+    """
+    log.debug('Fixing the opm path in index.Dockerfile')
+    dockerfile_path = os.path.join(dockerfile_dir, 'index.Dockerfile')
+    for line in fileinput.input(dockerfile_path, inplace=True):
+        print(line.replace('/build/bin/opm', '/bin/opm'), end='')
+
+
+def _get_local_pull_spec(request_id):
+    """
+    Get the pull specification of the index image for this request.
+
+    :return: the pull specification of the index image for this request.
+    :rtype: str
+    """
+    return f'operator-registry-index:{request_id}'
+
+
+def _push_arch_image(request_id):
+    """
+    Push the single arch index image to the configured registry.
+
+    :param int request_id: the ID of the IIB build request
+    :raises iib.exceptions.IIBError: if the push fails
+    """
+    source = _get_local_pull_spec(request_id)
+    conf = get_worker_config()
+    destination = conf['iib_arch_image_push_template'].format(
+        registry=conf['iib_registry'], request_id=request_id, arch=conf['iib_arch']
+    )
+    log.info('Pushing the index image %s to %s', source, destination)
+    _run_cmd(
+        ['podman', 'push', '-q', source, destination, '--creds', conf['iib_registry_credentials']],
+        exc_msg=f'Failed to push the index image to {destination}',
+    )
+
+
+@app.task
+def opm_index_add(bundles, binary_image, request_id, from_index=None):
+    """
+    Build and push an operator index image for a specific architecture with the input bundles.
+
+    :param list bundles: a list of strings representing the pull specifications of the bundles to
+        add to the index image being built.
+    :param str binary_image: the pull specification of the image where the opm binary gets copied
+        from.
+    :param int request_id: the ID of the IIB build request
+    :param str from_index: the pull specification of the image containing the index that the index
+        image build will be based from.
+    :raises iib.exceptions.IIBError: if the index image build fails.
+    """
+    _cleanup()
+    with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
+        # TODO: Once https://github.com/operator-framework/operator-registry/pull/173 is
+        # released, opm can just call podman directly for us.
+        cmd = [
+            'opm',
+            'index',
+            'add',
+            '--generate',
+            '--bundles',
+            ','.join(bundles),
+            '--binary-image',
+            binary_image,
+        ]
+
+        bundles_str = ', '.join(bundles)
+        if from_index:
+            log.info(
+                'Generating the database file from %s and adding the following bundle(s): %s',
+                from_index,
+                bundles_str,
+            )
+            cmd.extend(['--from-index', from_index])
+        else:
+            log.info(
```

Comment:
Can we log here that from_index wasn't specified?

Reply:
we can potentially simplify this if-else block by logging something like `log.info(
                'Generating the database file with the following bundle(s): %s'.,
                bundles_str,
            )` and then adding another log statement with the value of the `index_image` only if it is specified. That way we can eliminate the else block. does that make sense?